### PR TITLE
me/maine_gov normalize

### LINKED ISSUE #129: normalize maine_gov from Maine
[learn our pipeline: normalize](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Runner-pipeline-stages#normalize)

Blocked by #128 

Normalize existing [`.ndjson`](http://ndjson.org/) records into [the Vaccinate The States data schema](https://github.com/CAVaccineInventory/vaccine-feed-ingest/blob/main/vaccine_feed_ingest/schema/schema.py).

Read all files in the directory passed as the second argument (`sys.argv[2]`), transform each record into the normalized schema (filling in as much detail as possible), then output them to new files in the directory passed as the first argument (`sys.argv[1]`).

Check the wiki to learn more about the purpose of the normalize stage and how to get set up for development!

### Tips

1. Fetch and parse data for this site before you start developing:
    ```sh
    poetry run vaccine-feed-ingest fetch me/main_gov
    poetry run vaccine-feed-ingest parse me/main_gov
    ```

1. While working on your code, run it at any point:
    ```sh
    poetry run vaccine-feed-ingest normalize me/main_gov
    ```

### Example
[Parsed files for `ak/arcgis`](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/parsed) are converted to [normalized files](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/normalized)

### PR DESCRIPTION
**Pull request by rossry**
# normalize maine_gov for me

| Key Details |
|-|
| Resolves #129 |
State: me |
Site: maine_gov |

## Notes
I borked my git branches, so this one also includes il/sfsites/parse.yml, which was PR'd in #213. Probably just wait for that one to be accepted before considering this one. (Or someone more fluent in git than me can try to pull the il/sfsites/parse.yml out of this one, so they're properly independent.)
No street addresses, so city goes in the notes field.
Some locations have more than one phone number, so I included them all (with null website) and added the website in a different Contact item.

## Data sample

```json
{"id": "me:maine_gov:Bridgton Hospital:Bridgton", "name": "Bridgton Hospital", "address": null, "location": null, "contact": [{"contact_type": null, "phone": "207-520-2917", "website": null, "email": null, "other": null}, {"contact_type": null, "phone": null, "website": "https://www.cmhc.org/about-us/cmh-alert/vaccine-covid-19/", "email": null, "other": null}, {"contact_type": null, "phone": null, "website": null, "email": null, "other": "Call <a href=\"tel:207-520-2917\">207-520-2917</a> to make an appointment. <a href=\"https://www.cmhc.org/about-us/cmh-alert/vaccine-covid-19/\">More information here.</a>"}], "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": ["city:Bridgton", "county:Cumberland", "audience:Public", "info:Call <a href=\"tel:207-520-2917\">207-520-2917</a> to make an appointment. <a href=\"https://www.cmhc.org/about-us/cmh-alert/vaccine-covid-19/\">More information here.</a>"], "active": null, "source": {"source": "me:maine_gov", "id": "me:maine_gov:Bridgton Hospital:Bridgton", "fetched_from_uri": "https://www.maine.gov/covid19/vaccines/vaccination-sites", "fetched_at": "2021-04-26T07:51:18.665129", "published_at": null, "data": {"providerName": "Bridgton Hospital", "city": "Bridgton", "county": "Cumberland", "schedulingInfo": "Call <a href=\"tel:207-520-2917\">207-520-2917</a> to make an appointment. <a href=\"https://www.cmhc.org/about-us/cmh-alert/vaccine-covid-19/\">More information here.</a>", "minimumAge": "", "audience": "Public", "phoneNumber": ["207-520-2917"], "website": ["https://www.cmhc.org/about-us/cmh-alert/vaccine-covid-19/"]}}}
{"id": "me:maine_gov:Bucksport Regional Health Center:Bucksport", "name": "Bucksport Regional Health Center", "address": null, "location": null, "contact": [], "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": ["city:Bucksport", "county:Hancock", "audience:Patient Only", "info:By appointment only. Patients will be notified of eligibility to receive the vaccine."], "active": null, "source": {"source": "me:maine_gov", "id": "me:maine_gov:Bucksport Regional Health Center:Bucksport", "fetched_from_uri": "https://www.maine.gov/covid19/vaccines/vaccination-sites", "fetched_at": "2021-04-26T07:51:18.665129", "published_at": null, "data": {"providerName": "Bucksport Regional Health Center", "city": "Bucksport", "county": "Hancock", "schedulingInfo": "By appointment only. Patients will be notified of eligibility to receive the vaccine.", "minimumAge": "", "audience": "Patient Only", "phoneNumber": [], "website": []}}}
{"id": "me:maine_gov:B Street Clinic:Lewiston", "name": "B Street Clinic", "address": null, "location": null, "contact": [{"contact_type": null, "phone": null, "website": "http://www.communityclinicalservices.com/covid-19-vaccines/", "email": null, "other": null}, {"contact_type": null, "phone": null, "website": null, "email": null, "other": "To make an appointment visit <a href=\"http://www.communityclinicalservices.com/covid-19-vaccines/\">http://www.communityclinicalservices.com/covid-19-vaccines/</a>."}], "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": ["city:Lewiston", "county:Androscoggin", "audience:Public", "info:To make an appointment visit <a href=\"http://www.communityclinicalservices.com/covid-19-vaccines/\">http://www.communityclinicalservices.com/covid-19-vaccines/</a>."], "active": null, "source": {"source": "me:maine_gov", "id": "me:maine_gov:B Street Clinic:Lewiston", "fetched_from_uri": "https://www.maine.gov/covid19/vaccines/vaccination-sites", "fetched_at": "2021-04-26T07:51:18.665129", "published_at": null, "data": {"providerName": "B Street Clinic", "city": "Lewiston", "county": "Androscoggin", "schedulingInfo": "To make an appointment visit <a href=\"http://www.communityclinicalservices.com/covid-19-vaccines/\">http://www.communityclinicalservices.com/covid-19-vaccines/</a>.", "minimumAge": "", "audience": "Public", "phoneNumber": [], "website": ["http://www.communityclinicalservices.com/covid-19-vaccines/"]}}}
{"id": "me:maine_gov:Calais Regional Hospital:Calais", "name": "Calais Regional Hospital", "address": null, "location": null, "contact": [{"contact_type": null, "phone": null, "website": "https://www.calaishospital.org/covid-19/covid-19-vaccination/covid-19-vaccine-appointment-registration-page/", "email": null, "other": null}, {"contact_type": null, "phone": null, "website": null, "email": null, "other": "\n<p>Visit <a href=\"https://www.calaishospital.org/covid-19/covid-19-vaccination/covid-19-vaccine-appointment-registration-page/\">www.calaishospital.org/covid-19/</a> to register for an appointment.</p>\n"}], "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": ["city:Calais", "county:Washington County", "audience:Public", "info:\n<p>Visit <a href=\"https://www.calaishospital.org/covid-19/covid-19-vaccination/covid-19-vaccine-appointment-registration-page/\">www.calaishospital.org/covid-19/</a> to register for an appointment.</p>\n"], "active": null, "source": {"source": "me:maine_gov", "id": "me:maine_gov:Calais Regional Hospital:Calais", "fetched_from_uri": "https://www.maine.gov/covid19/vaccines/vaccination-sites", "fetched_at": "2021-04-26T07:51:18.665129", "published_at": null, "data": {"providerName": "Calais Regional Hospital", "city": "Calais", "county": "Washington County", "schedulingInfo": "\n<p>Visit <a href=\"https://www.calaishospital.org/covid-19/covid-19-vaccination/covid-19-vaccine-appointment-registration-page/\">www.calaishospital.org/covid-19/</a> to register for an appointment.</p>\n", "minimumAge": "", "audience": "Public", "phoneNumber": [], "website": ["https://www.calaishospital.org/covid-19/covid-19-vaccination/covid-19-vaccine-appointment-registration-page/"]}}}
```

## Before Opening a PR
- [x] I tested this using the CLI (e.g., `poetry run vaccine-feed-ingest <state>/<site>`)
- [x] I ran auto-formatting:
  - [x] `poetry run black vaccine_feed_ingest/runners`
  - [x] `poetry run isort vaccine_feed_ingest/runners`

### CODE REVIEW

File: vaccine_feed_ingest/runners/me/maine_gov/normalize.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        ret.append(schema.Contact(phone=phone_number))
+    for website in site["website"]:
+        ret.append(schema.Contact(website=website))
+        ret.append(schema.Contact(other=site["schedulingInfo"]))
+    return ret
+
+
+def _get_notes(site: dict):
+    ret = []
+    ret.append("city:" + site["city"])
```
Comment (tmb) [inline vaccine_feed_ingest/runners/me/maine_gov/normalize.py:58]: @eliblock I wonder how we can include this data in the schema without resorting to notes. These Maine sites have city & county info but not zip/street1 so they don't fit into either of the location objects right now.

File: vaccine_feed_ingest/runners/me/maine_gov/normalize.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        fetched_from_uri="https://www.maine.gov/covid19/vaccines/vaccination-sites",
+        id=_get_id(site),
+        source="me:maine_gov",
+    )
+
+
+def _get_contacts(site: dict):
+    ret = []
+    for phone_number in site["phoneNumber"]:
+        ret.append(schema.Contact(phone=phone_number))
```
Comment (tmb) [inline vaccine_feed_ingest/runners/me/maine_gov/normalize.py:49]: Could you format the phone number to be in this format: `(xxx) xxx-xxxx`? Also, this contact can probably have a `contact_type` of `general`.

Comment (tmb) [reply vaccine_feed_ingest/runners/me/maine_gov/normalize.py:49]: You also can probably jam the website into this object too.

**Review by tmb — CHANGES_REQUESTED** [review]
Thanks for submitting a PR! There are just a few small things to fix (noted in-line) but should be good to merge shortly :)

File: vaccine_feed_ingest/runners/me/maine_gov/normalize.py (hunk 1 above)
Comment (eliblock) [reply vaccine_feed_ingest/runners/me/maine_gov/normalize.py:58]: I _think_ the plan is to have a data-cleaning process add this info later in the pipeline if it's not available. @bryanculbertson - could you share some guidance here?

File: vaccine_feed_ingest/runners/me/maine_gov/normalize.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    city = _get_city(site)
+    return f"me:maine_gov:{name}:{city}"
+
+
+def _get_source(site: dict, timestamp: str) -> schema.Source:
+    return schema.Source(
+        data=site,
+        fetched_at=timestamp,
+        fetched_from_uri="https://www.maine.gov/covid19/vaccines/vaccination-sites",
+        id=_get_id(site),
```
Comment (tmb) [inline vaccine_feed_ingest/runners/me/maine_gov/normalize.py:41]: Forgot one thing, can you have the ID (just in Source) match with whatever record ID the maine page gives? if it doesn't specify anything, feel free to conjoin the name and city like you did above without the `me:maine_gov` namespace :)

File: vaccine_feed_ingest/runners/me/maine_gov/normalize.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        source="me:maine_gov",
+    )
+
+
+def _get_contacts(site: dict):
+    ret = []
+    for phone_number in site["phoneNumber"]:
+        ret.append(schema.Contact(phone=phone_number))
+    for website in site["website"]:
+        ret.append(schema.Contact(website=website))
```
Comment (tmb) [inline vaccine_feed_ingest/runners/me/maine_gov/normalize.py:51]: This Contact object can probably have a `contact_type` of `general`

**Comment by rossry** [comment]
Thanks for the comments; should be able to address after work in ~~8 hours.

File: vaccine_feed_ingest/runners/me/maine_gov/normalize.py (hunk 2 above)
Comment (rossry) [reply vaccine_feed_ingest/runners/me/maine_gov/normalize.py:49]: Realized that phone phone numbers are missing area code (they're 7 digits in "XXX-XXXX" format). What do we want to do with them?

I'll push "(???) XXX-XXXX" with question marks for now.

File: vaccine_feed_ingest/runners/me/maine_gov/normalize.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        ret.append(schema.Contact(other=site["schedulingInfo"]))
```
[the remark this answers is not in the document]
Comment (rossry) [reply vaccine_feed_ingest/runners/me/maine_gov/normalize.py]: Did this, and also tried to pull out phone and website when they could be found in the schedulingInfo.

File: vaccine_feed_ingest/runners/me/maine_gov/normalize.py (hunk 2 above)
Comment (rossry) [reply vaccine_feed_ingest/runners/me/maine_gov/normalize.py]: Done, with the above representation for 7-digit phone numbers.

File: vaccine_feed_ingest/runners/me/maine_gov/normalize.py (hunk 4 above)
Comment (rossry) [reply vaccine_feed_ingest/runners/me/maine_gov/normalize.py]: Done.

File: vaccine_feed_ingest/runners/me/maine_gov/normalize.py (hunk 3 above)
Comment (rossry) [reply vaccine_feed_ingest/runners/me/maine_gov/normalize.py:41]: Done.

File: vaccine_feed_ingest/runners/me/maine_gov/normalize.py (hunk 1 above)
Comment (rossry) [reply vaccine_feed_ingest/runners/me/maine_gov/normalize.py:116]: Do we have a resolution here?

If not, do we think this issue should be blocking?

**Comment by rossry** [comment]
Latest output sample:
```{json}
{"id": "me:maine_gov:Bridgton Hospital:Bridgton", "name": "Bridgton Hospital", "address": null, "location": null, "contact": [{"contact_type": "general", "phone": "(207) 520-2917", "website": null, "email": null, "other": null}, {"contact_type": "general", "phone": null, "website": "https://www.cmhc.org/about-us/cmh-alert/vaccine-covid-19/", "email": null, "other": null}, {"contact_type": "booking", "phone": "(207) 520-2917", "website": "https://www.cmhc.org/about-us/cmh-alert/vaccine-covid-19/", "email": null, "other": "Call <a href=\"tel:207-520-2917\">207-520-2917</a> to make an appointment. <a href=\"https://www.cmhc.org/about-us/cmh-alert/vaccine-covid-19/\">More information here.</a>"}], "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": ["city:Bridgton", "county:Cumberland", "audience:Public", "info:Call <a href=\"tel:207-520-2917\">207-520-2917</a> to make an appointment. <a href=\"https://www.cmhc.org/about-us/cmh-alert/vaccine-covid-19/\">More information here.</a>"], "active": null, "source": {"source": "me:maine_gov", "id": "Bridgton Hospital:Bridgton", "fetched_from_uri": "https://www.maine.gov/covid19/vaccines/vaccination-sites", "fetched_at": "2021-04-27T15:04:50.736368", "published_at": null, "data": {"providerName": "Bridgton Hospital", "city": "Bridgton", "county": "Cumberland", "schedulingInfo": "Call <a href=\"tel:207-520-2917\">207-520-2917</a> to make an appointment. <a href=\"https://www.cmhc.org/about-us/cmh-alert/vaccine-covid-19/\">More information here.</a>", "minimumAge": "", "audience": "Public", "phoneNumber": ["207-520-2917"], "website": ["https://www.cmhc.org/about-us/cmh-alert/vaccine-covid-19/"]}}}
{"id": "me:maine_gov:Bucksport Regional Health Center:Bucksport", "name": "Bucksport Regional Health Center", "address": null, "location": null, "contact": [{"contact_type": "booking", "phone": null, "website": null, "email": null, "other": "By appointment only. Patients will be notified of eligibility to receive the vaccine."}], "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": ["city:Bucksport", "county:Hancock", "audience:Patient Only", "info:By appointment only. Patients will be notified of eligibility to receive the vaccine."], "active": null, "source": {"source": "me:maine_gov", "id": "Bucksport Regional Health Center:Bucksport", "fetched_from_uri": "https://www.maine.gov/covid19/vaccines/vaccination-sites", "fetched_at": "2021-04-27T15:04:50.736368", "published_at": null, "data": {"providerName": "Bucksport Regional Health Center", "city": "Bucksport", "county": "Hancock", "schedulingInfo": "By appointment only. Patients will be notified of eligibility to receive the vaccine.", "minimumAge": "", "audience": "Patient Only", "phoneNumber": [], "website": []}}}
{"id": "me:maine_gov:B Street Clinic:Lewiston", "name": "B Street Clinic", "address": null, "location": null, "contact": [{"contact_type": "general", "phone": null, "website": "http://www.communityclinicalservices.com/covid-19-vaccines/", "email": null, "other": null}, {"contact_type": "booking", "phone": null, "website": "http://www.communityclinicalservices.com/covid-19-vaccines/", "email": null, "other": "To make an appointment visit <a href=\"http://www.communityclinicalservices.com/covid-19-vaccines/\">http://www.communityclinicalservices.com/covid-19-vaccines/</a>."}], "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": ["city:Lewiston", "county:Androscoggin", "audience:Public", "info:To make an appointment visit <a href=\"http://www.communityclinicalservices.com/covid-19-vaccines/\">http://www.communityclinicalservices.com/covid-19-vaccines/</a>."], "active": null, "source": {"source": "me:maine_gov", "id": "B Street Clinic:Lewiston", "fetched_from_uri": "https://www.maine.gov/covid19/vaccines/vaccination-sites", "fetched_at": "2021-04-27T15:04:50.736368", "published_at": null, "data": {"providerName": "B Street Clinic", "city": "Lewiston", "county": "Androscoggin", "schedulingInfo": "To make an appointment visit <a href=\"http://www.communityclinicalservices.com/covid-19-vaccines/\">http://www.communityclinicalservices.com/covid-19-vaccines/</a>.", "minimumAge": "", "audience": "Public", "phoneNumber": [], "website": ["http://www.communityclinicalservices.com/covid-19-vaccines/"]}}}
{"id": "me:maine_gov:Calais Regional Hospital:Calais", "name": "Calais Regional Hospital", "address": null, "location": null, "contact": [{"contact_type": "general", "phone": null, "website": "https://www.calaishospital.org/covid-19/covid-19-vaccination/covid-19-vaccine-appointment-registration-page/", "email": null, "other": null}, {"contact_type": "booking", "phone": null, "website": "https://www.calaishospital.org/covid-19/covid-19-vaccination/covid-19-vaccine-appointment-registration-page/", "email": null, "other": "\n<p>Visit <a href=\"https://www.calaishospital.org/covid-19/covid-19-vaccination/covid-19-vaccine-appointment-registration-page/\">www.calaishospital.org/covid-19/</a> to register for an appointment.</p>\n"}], "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": ["city:Calais", "county:Washington County", "audience:Public", "info:\n<p>Visit <a href=\"https://www.calaishospital.org/covid-19/covid-19-vaccination/covid-19-vaccine-appointment-registration-page/\">www.calaishospital.org/covid-19/</a> to register for an appointment.</p>\n"], "active": null, "source": {"source": "me:maine_gov", "id": "Calais Regional Hospital:Calais", "fetched_from_uri": "https://www.maine.gov/covid19/vaccines/vaccination-sites", "fetched_at": "2021-04-27T15:04:50.736368", "published_at": null, "data": {"providerName": "Calais Regional Hospital", "city": "Calais", "county": "Washington County", "schedulingInfo": "\n<p>Visit <a href=\"https://www.calaishospital.org/covid-19/covid-19-vaccination/covid-19-vaccine-appointment-registration-page/\">www.calaishospital.org/covid-19/</a> to register for an appointment.</p>\n", "minimumAge": "", "audience": "Public", "phoneNumber": [], "website": ["https://www.calaishospital.org/covid-19/covid-19-vaccination/covid-19-vaccine-appointment-registration-page/"]}}}
{"id": "me:maine_gov:Caribou Health Center:Caribou", "name": "Caribou Health Center", "address": null, "location": null, "contact": [{"contact_type": "general", "phone": "(207) 498-1638", "website": null, "email": null, "other": null}, {"contact_type": "general", "phone": null, "website": "http://www.pineshealth.org/", "email": null, "other": null}, {"contact_type": "booking", "phone": "(207) 498-1638", "website": "http://www.pineshealth.org/", "email": null, "other": "Information on vaccination clinics provided by Pines Health Services is available by calling <a href=\"tel:207-498-1638\">207-498-1638</a> or visiting <a href=\"http://www.pineshealth.org/\">www.pineshealth.org/</a>."}], "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": ["city:Caribou", "county:Aroostook", "audience:Patient Only", "info:Information on vaccination clinics provided by Pines Health Services is available by calling <a href=\"tel:207-498-1638\">207-498-1638</a> or visiting <a href=\"http://www.pineshealth.org/\">www.pineshealth.org/</a>."], "active": null, "source": {"source": "me:maine_gov", "id": "Caribou Health Center:Caribou", "fetched_from_uri": "https://www.maine.gov/covid19/vaccines/vaccination-sites", "fetched_at": "2021-04-27T15:04:50.736368", "published_at": null, "data": {"providerName": "Caribou Health Center", "city": "Caribou", "county": "Aroostook", "schedulingInfo": "Information on vaccination clinics provided by Pines Health Services is available by calling <a href=\"tel:207-498-1638\">207-498-1638</a> or visiting <a href=\"http://www.pineshealth.org/\">www.pineshealth.org/</a>.", "minimumAge": "", "audience": "Patient Only", "phoneNumber": ["207-498-1638"], "website": ["http://www.pineshealth.org/"]}}}
{"id": "me:maine_gov:Cary Medical Center:Caribou", "name": "Cary Medical Center", "address": null, "location": null, "contact": [{"contact_type": "general", "phone": "(207) 498-1638", "website": null, "email": null, "other": null}, {"contact_type": "general", "phone": null, "website": "http://www.carymedicalcenter.org/covid-19/", "email": null, "other": null}, {"contact_type": "booking", "phone": "(207) 498-1638", "website": "http://www.carymedicalcenter.org/covid-19/", "email": null, "other": "Call <a href=\"tel:207-498-1638\">207-498-1638</a> or visit<a href=\"http://www.carymedicalcenter.org/covid-19/\">www.carymedicalcenter.org/covid-19/</a> to make an appointment."}], "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": ["city:Caribou", "county:Aroostook", "audience:Public", "info:Call <a href=\"tel:207-498-1638\">207-498-1638</a> or visit<a href=\"http://www.carymedicalcenter.org/covid-19/\">www.carymedicalcenter.org/covid-19/</a> to make an appointment."], "active": null, "source": {"source": "me:maine_gov", "id": "Cary Medical Center:Caribou", "fetched_from_uri": "https://www.maine.gov/covid19/vaccines/vaccination-sites", "fetched_at": "2021-04-27T15:04:50.736368", "published_at": null, "data": {"providerName": "Cary Medical Center", "city": "Caribou", "county": "Aroostook", "schedulingInfo": "Call <a href=\"tel:207-498-1638\">207-498-1638</a> or visit<a href=\"http://www.carymedicalcenter.org/covid-19/\">www.carymedicalcenter.org/covid-19/</a> to make an appointment.", "minimumAge": "", "audience": "Public", "phoneNumber": ["207-498-1638"], "website": ["http://www.carymedicalcenter.org/covid-19/"]}}}
```

**Comment by rossry** [comment]
Latest output sample:
```{json}
{"id": "me:maine_gov:Bridgton Hospital:Bridgton", "name": "Bridgton Hospital", "address": null, "location": null, "contact": [{"contact_type": "general", "phone": "(207) 520-2917", "website": null, "email": null, "other": null}, {"contact_type": "general", "phone": null, "website": "https://www.cmhc.org/about-us/cmh-alert/vaccine-covid-19/", "email": null, "other": null}, {"contact_type": "booking", "phone": "(207) 520-2917", "website": "https://www.cmhc.org/about-us/cmh-alert/vaccine-covid-19/", "email": null, "other": "Call <a href=\"tel:207-520-2917\">207-520-2917</a> to make an appointment. <a href=\"https://www.cmhc.org/about-us/cmh-alert/vaccine-covid-19/\">More information here.</a>"}], "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": ["city:Bridgton", "county:Cumberland", "audience:Public"], "active": null, "source": {"source": "me:maine_gov", "id": "Bridgton Hospital:Bridgton", "fetched_from_uri": "https://www.maine.gov/covid19/vaccines/vaccination-sites", "fetched_at": "2021-04-27T15:07:00.264386", "published_at": null, "data": {"providerName": "Bridgton Hospital", "city": "Bridgton", "county": "Cumberland", "schedulingInfo": "Call <a href=\"tel:207-520-2917\">207-520-2917</a> to make an appointment. <a href=\"https://www.cmhc.org/about-us/cmh-alert/vaccine-covid-19/\">More information here.</a>", "minimumAge": "", "audience": "Public", "phoneNumber": ["207-520-2917"], "website": ["https://www.cmhc.org/about-us/cmh-alert/vaccine-covid-19/"]}}}
{"id": "me:maine_gov:Bucksport Regional Health Center:Bucksport", "name": "Bucksport Regional Health Center", "address": null, "location": null, "contact": [{"contact_type": "booking", "phone": null, "website": null, "email": null, "other": "By appointment only. Patients will be notified of eligibility to receive the vaccine."}], "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": ["city:Bucksport", "county:Hancock", "audience:Patient Only"], "active": null, "source": {"source": "me:maine_gov", "id": "Bucksport Regional Health Center:Bucksport", "fetched_from_uri": "https://www.maine.gov/covid19/vaccines/vaccination-sites", "fetched_at": "2021-04-27T15:07:00.264386", "published_at": null, "data": {"providerName": "Bucksport Regional Health Center", "city": "Bucksport", "county": "Hancock", "schedulingInfo": "By appointment only. Patients will be notified of eligibility to receive the vaccine.", "minimumAge": "", "audience": "Patient Only", "phoneNumber": [], "website": []}}}
{"id": "me:maine_gov:B Street Clinic:Lewiston", "name": "B Street Clinic", "address": null, "location": null, "contact": [{"contact_type": "general", "phone": null, "website": "http://www.communityclinicalservices.com/covid-19-vaccines/", "email": null, "other": null}, {"contact_type": "booking", "phone": null, "website": "http://www.communityclinicalservices.com/covid-19-vaccines/", "email": null, "other": "To make an appointment visit <a href=\"http://www.communityclinicalservices.com/covid-19-vaccines/\">http://www.communityclinicalservices.com/covid-19-vaccines/</a>."}], "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": ["city:Lewiston", "county:Androscoggin", "audience:Public"], "active": null, "source": {"source": "me:maine_gov", "id": "B Street Clinic:Lewiston", "fetched_from_uri": "https://www.maine.gov/covid19/vaccines/vaccination-sites", "fetched_at": "2021-04-27T15:07:00.264386", "published_at": null, "data": {"providerName": "B Street Clinic", "city": "Lewiston", "county": "Androscoggin", "schedulingInfo": "To make an appointment visit <a href=\"http://www.communityclinicalservices.com/covid-19-vaccines/\">http://www.communityclinicalservices.com/covid-19-vaccines/</a>.", "minimumAge": "", "audience": "Public", "phoneNumber": [], "website": ["http://www.communityclinicalservices.com/covid-19-vaccines/"]}}}
{"id": "me:maine_gov:Calais Regional Hospital:Calais", "name": "Calais Regional Hospital", "address": null, "location": null, "contact": [{"contact_type": "general", "phone": null, "website": "https://www.calaishospital.org/covid-19/covid-19-vaccination/covid-19-vaccine-appointment-registration-page/", "email": null, "other": null}, {"contact_type": "booking", "phone": null, "website": "https://www.calaishospital.org/covid-19/covid-19-vaccination/covid-19-vaccine-appointment-registration-page/", "email": null, "other": "\n<p>Visit <a href=\"https://www.calaishospital.org/covid-19/covid-19-vaccination/covid-19-vaccine-appointment-registration-page/\">www.calaishospital.org/covid-19/</a> to register for an appointment.</p>\n"}], "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": ["city:Calais", "county:Washington County", "audience:Public"], "active": null, "source": {"source": "me:maine_gov", "id": "Calais Regional Hospital:Calais", "fetched_from_uri": "https://www.maine.gov/covid19/vaccines/vaccination-sites", "fetched_at": "2021-04-27T15:07:00.264386", "published_at": null, "data": {"providerName": "Calais Regional Hospital", "city": "Calais", "county": "Washington County", "schedulingInfo": "\n<p>Visit <a href=\"https://www.calaishospital.org/covid-19/covid-19-vaccination/covid-19-vaccine-appointment-registration-page/\">www.calaishospital.org/covid-19/</a> to register for an appointment.</p>\n", "minimumAge": "", "audience": "Public", "phoneNumber": [], "website": ["https://www.calaishospital.org/covid-19/covid-19-vaccination/covid-19-vaccine-appointment-registration-page/"]}}}
{"id": "me:maine_gov:Caribou Health Center:Caribou", "name": "Caribou Health Center", "address": null, "location": null, "contact": [{"contact_type": "general", "phone": "(207) 498-1638", "website": null, "email": null, "other": null}, {"contact_type": "general", "phone": null, "website": "http://www.pineshealth.org/", "email": null, "other": null}, {"contact_type": "booking", "phone": "(207) 498-1638", "website": "http://www.pineshealth.org/", "email": null, "other": "Information on vaccination clinics provided by Pines Health Services is available by calling <a href=\"tel:207-498-1638\">207-498-1638</a> or visiting <a href=\"http://www.pineshealth.org/\">www.pineshealth.org/</a>."}], "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": ["city:Caribou", "county:Aroostook", "audience:Patient Only"], "active": null, "source": {"source": "me:maine_gov", "id": "Caribou Health Center:Caribou", "fetched_from_uri": "https://www.maine.gov/covid19/vaccines/vaccination-sites", "fetched_at": "2021-04-27T15:07:00.264386", "published_at": null, "data": {"providerName": "Caribou Health Center", "city": "Caribou", "county": "Aroostook", "schedulingInfo": "Information on vaccination clinics provided by Pines Health Services is available by calling <a href=\"tel:207-498-1638\">207-498-1638</a> or visiting <a href=\"http://www.pineshealth.org/\">www.pineshealth.org/</a>.", "minimumAge": "", "audience": "Patient Only", "phoneNumber": ["207-498-1638"], "website": ["http://www.pineshealth.org/"]}}}
{"id": "me:maine_gov:Cary Medical Center:Caribou", "name": "Cary Medical Center", "address": null, "location": null, "contact": [{"contact_type": "general", "phone": "(207) 498-1638", "website": null, "email": null, "other": null}, {"contact_type": "general", "phone": null, "website": "http://www.carymedicalcenter.org/covid-19/", "email": null, "other": null}, {"contact_type": "booking", "phone": "(207) 498-1638", "website": "http://www.carymedicalcenter.org/covid-19/", "email": null, "other": "Call <a href=\"tel:207-498-1638\">207-498-1638</a> or visit<a href=\"http://www.carymedicalcenter.org/covid-19/\">www.carymedicalcenter.org/covid-19/</a> to make an appointment."}], "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": ["city:Caribou", "county:Aroostook", "audience:Public"], "active": null, "source": {"source": "me:maine_gov", "id": "Cary Medical Center:Caribou", "fetched_from_uri": "https://www.maine.gov/covid19/vaccines/vaccination-sites", "fetched_at": "2021-04-27T15:07:00.264386", "published_at": null, "data": {"providerName": "Cary Medical Center", "city": "Caribou", "county": "Aroostook", "schedulingInfo": "Call <a href=\"tel:207-498-1638\">207-498-1638</a> or visit<a href=\"http://www.carymedicalcenter.org/covid-19/\">www.carymedicalcenter.org/covid-19/</a> to make an appointment.", "minimumAge": "", "audience": "Public", "phoneNumber": ["207-498-1638"], "website": ["http://www.carymedicalcenter.org/covid-19/"]}}}
```

File: vaccine_feed_ingest/runners/me/maine_gov/normalize.py (hunk 1 above)
Comment (tmb) [reply vaccine_feed_ingest/runners/me/maine_gov/normalize.py:117]: Not yet, I think. Definitely shouldn't block—we can patch it up in a future PR if the schema changes to allow.

**Comment by tmb** [comment]
🎉 merging now — thank you so much for the PR!

just made a few changes to the contact objects (w a new way we're structuring them, having one contact method per object) & phone formatting (because of those wonky area codes).

**Pull request merged**